ERCS/erc-7828.md

```diff
@@ -0,0 +1,236 @@
+
+## Motivation
+
+The current Ethereum address landscape is leading to an ecosystem that will have hundreds and eventually thousands of L2s that use the same address format as Ethereum mainnet. This means an address by itself is not enough information to know which chain the address is related to. This can be problematic if funds are sent to an unreachable address on the incorrect chain. From the user account it should be possible to obtain the right chain identifier (chainID) to include in a transaction. 
+
+The mapping from chain names to identifiers has, since [EIP-155](./eip-155.md), been maintained off chain using a centralized list. This solution has two main shortcomings:
+ - It does not scale with the growing number of L2s.
+ - The list maintainer is a trusted centralized entity.
+
+Instead of using chain identifiers, which are not human readable, the address could be extended with a human-readable chain name, which can then be resolved to a chain identifier.
+The mapping from chain names to identifiers can be resolved off-chain using existing centralized lists or on-chain using ENS (see ERC-7785).
+
+In the same spirit, the address could be a human-readable name as well, which is already a use case for ENS. However it would be desirable if the address name could be registered on a L2.
+
+Desired properties:
+- a unified format to represent any address on L1 or L2
+- the ability to use chain names in addition to identifiers
+- the chain portion can be a domain name, or just the suffix for a "base chain" (eg. `eth`, `myfavoriterollup.eth`, `sepolia`, `my_l3.base.superchain.eth`)
+- the address portion can be either the appropriate type of address for the chain (0x... for EVM chains, otherwise eg. for starknet something else), or a domain name (ENS or other)
+- the ability to resolve ENS names on the L2
+- the address portion and the chain portion should be resolved separately 
+- checksums are MANDATORY
+- checksum hash goes over the entire address, so users can't just replace a component and expect it to stay valid
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Format
+
+Valid addresses MUST include the identifier of the chain they belong to.
+
+```
+L1-TLD ::= eth | sepolia | …
+chain_id ::= 0x[a-fA-F0-9]{1,64}
+address ::= 0x[a-fA-F0-9]{40}
+chain ::= <chain_id> | <L1-TLD> | <ens-name> . <L1-TLD>
+user ::= <address> | <ens-subdomain>
+account ::= <user>@<chain>
+```
+
+Note the difference between `ens-name`, which is a full ENS name, and `ens-subdomain` that is just a segment of a name between dots. E.g. `user.app.eth` is a name, `user` and `app` are subdomains.
+
+A few examples below. 
+
+Option 1: using @ to separate address and chain
```

Comment:
I think option 1 is better both aesthetically and verbally, its intuitive to say "at" a location which will make it easier for people with less context to understand